src/main/java/eu/tornplayground/tornapi/limiter/RequestLimiter.java

```diff
@@ -0,0 +1,71 @@
+package eu.tornplayground.tornapi.limiter;
+
+import java.util.Deque;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentLinkedDeque;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Limits the number of requests per given time frame per API key.
+ */
+public class RequestLimiter {
```

Comment:
I feel like this really should be an interface and this logic be moved to it's own class. This implementation spaces requests out evenly, but a different implementation could allow some more bursts of requests.

Reply:
You can still do bursts of requests tho.. if you have 100 requests per minute you can do 100 requests within the first second.. that doesn't matter.. it literally just checks if there has been X request within the given timeframe. But maybe i don't understand what you mean

Reply:
Oh yeah, now I understand. Was too fixated on the constructor for some reason.